docs/configuration.adoc

```diff
@@ -195,183 +195,219 @@ The following script levels are reserved by pre-defined scripts:
-<79> The username used for authentication.
-<80> The password used for authentication.
-<81> The authentication database, if different from the storage database.
-<82> The required replica set name.
-<83> Flag that indicates if an SSL connection should be used.
-<84> Flag that indicates if socket keep-alive should be enabled for connections to MongoDB.
\ No newline at end of file
+<1> The Key-Value Storage type.
+<2> The configuration for a MapDB key-value storage.
+<3> The directory for a MapDB key-value storage.
+<4> The configuration for the blob store.
+<5> The type of blob store. NOTE: The local blob storage should only be used with a single server and single scheduler running on the same machine.
+<6> Configuration for a file-based blob store.
+<7> The directory where the blobs are created.
+<8> Example configuration for an AWS cloud-native implementation. AWS uses DynamoDB for key-value storage and S3 for blob storage.
+<9> The AWS region. If not specified, it will use the default region discovery of the AWS SDK.
+<10> The AWS access key. If not specified, it will use the default credential discovery of the AWS SDK.
+<11> The AWS secret key. If not specified, it will use the default credential discovery of the AWS SDK.
+<12> The AWS DynamoDB table name.
+<13> The AWS DynamoDB provisioned read throughput.
+<14> The AWS DynamoDB provisioned write throughput.
+<15> Configuration for the local file cache. Required.
+<16> The configuration for an S3 blob store.
+<17> The S3 region. If not specified, it will use the default region discovery of the AWS SDK.
+<18> The AWS S3 access key. If not specified, it will use the default credential discovery of the AWS SDK.
+<19> The AWS S3 secret key. If not specified, it will use the default credential discovery of the AWS SDK.
+<20> The AWS S3 bucket name.
+<21> Example configuration for Azure cloud-native implementation. Azure uses CosmosDB for key-value storage and Azure Blob for blob storage
+<22> The CosmosDB  account host.
+<23> The CosmosDB account key.
+<24> The CosmosDB database.
+<25> The CosmosDB container name.
+<26> The CosmosDB throughput.
+<27> The root directory of the blob cache. Required.
+<28> The configuration for an Azure blob store.
+<29> The Azure connection string.
+<30> The Azure blob storage container name. Required if enabled.
+<31> Example configuration for Google cloud-native implementation. Google uses Firestore for key-value storage and GCS for blob storage.
+<32> The path to the Google Firestore Service Account File.
+<33> The Google Firestore collection.
+<34> The root directory of the blob cache. Required.
+<35> The configuration for the GCS blob store.
+<36> The path to the GCS Service Account File.
+<37> The GCS bucket name.
+<38> Example configuration for MongoDB key-value storage.
+<39> The MongoDB hostname and ports.
+<40> The name of the MongoDB database. Required if Mongo is used.
+<41> The name of the MongoDB collection.
```

Comment:
Missing the following settings for MongoDB:

- authDatabase
- sslEnabled
- replicaSet